api/app/events/status.py

```diff
@@ -0,0 +1,82 @@
+from app.events.models import Event
+from app.users.models import AppUser
+from app.invitedGuest.repository import InvitedGuestRepository as invited_guest_repository
+from app.responses.repository import ResponseRepository as response_repository
+from app.registrationResponse.repository import RegistrationRepository as registration_repository
+from app.registration.repository import OfferRepository as offer_repository
+from app.outcome.repository import OutcomeRepository as outcome_repository
+
+
+class EventStatus():
+    """Represents the status of a user at an event."""
+    def __init__(self, 
+                 invited_guest=None, 
+                 application_status=None,
+                 outcome_status=None,
+                 offer_status=None, 
+                 registration_status=None):
+        if invited_guest is None and registration_status is not None and offer_status is None:
+            raise ValueError('Non-invited guest must have an offer to have a registration status')
+        
+        if invited_guest is None and outcome_status is not None and application_status is None:
```

Comment:
Can an outcome_status be not None, while an application_status is None?

Reply:
Logically it shouldn't be (hence the check), but technically there is nothing (eg DB constraints) preventing that from happening.